OsmAnd-java/src/main/java/net/osmand/search/core/SearchPhrase.java

```diff
@@ -209,19 +212,56 @@ private SearchPhrase createNewSearchPhrase(SearchSettings settings, String fullT
 				String wd = ws[i].trim();
 				boolean conjunction = conjunctions.contains(wd.toLowerCase());
 				boolean lastAndIncomplete = i == ws.length - 1 && !sp.lastUnknownSearchWordComplete;
+				boolean decryptAbbreviations = needDecryptAbbreviations(getWorldRegions());
+
 				if (wd.length() > 0 && (!conjunction || lastAndIncomplete)) {
 					if (first) {
-						sp.firstUnknownSearchWord = Abbreviations.replace(wd);
+						sp.firstUnknownSearchWord = decryptAbbreviations ? Abbreviations.replace(wd) : wd;
 						first = false;
 					} else {
-						sp.otherUnknownWords.add(Abbreviations.replace(wd));
+						sp.otherUnknownWords.add(decryptAbbreviations ? Abbreviations.replace(wd) : wd);
 					}
 				}
 			}
 		}
 		return sp;
 	}
-	
+
+	private List<WorldRegion> getWorldRegions() {
```

Comment:
Doesn't look good won't work on Android, it works differently probably WorldRegion should be passed by Android